test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -669,6 +670,11 @@ public static void assertVersionSerializable(Version version, Streamable streama
                     equalTo(0));
             assertThat("Serialization failed with version [" + version + "] bytes should be equal for streamable [" + streamable + "]",
                     serialize(version, streamable), equalTo(orig));
+        } catch (UnsupportedOperationException ex) {
+            if (ex.equals("can't read named writeable from StreamInput")) {
```

Comment:
Hmmm, I see. Is there no way for us to provide the NamedWriteableRegistry here? I think it's important that we are able to perform this check for the aggregation objects to ensure we don't accidentally introduce breaking changes (this method has saved me more than once).

Also can we re-throw the exception (probably wrapped in a Runtime exception as below) if the `if` statement fails rather than just swallowing the exception?